Don't chomp context in componentWillReceiveProps

alexf101 opened the pull request:
React's poor documentation for context is probably to blame here.

When a component defines 'contextTypes', componentWillReceiveProps also receives a nextContext parameter, which is otherwise null.

Without this patch, components that implement both context and componentWillReceiveProps stop receiving that extra parameter when strictPropTypes is in use.

alexf101 committed:
Don't chomp context in componentWillReceiveProps

React's poor documentation for context is probably to blame here.

When a component defines 'contextTypes', componentWillReceiveProps also receives a nextContext parameter, which is otherwise null.

Without this patch, components that implement both context and componentWillReceiveProps stop receiving that extra parameter when strictPropTypes is in use.

gajus commented:
> I don't really get how releasing this is meant to work, but copying it to dist seems to be what's happened in the past.

Please remove the commit that includes `./dist` in the repository. Packages are published manually.

alexf101 committed:
Remove lodash as a run dependency.

This package should be tiny and doesn't need dependencies. Keep it
as a dev dependency though for pragmatist.

alexf101 commented:
@gajus sorry for the mistake and long delay getting back to you. My email filters were set way too aggressive for GitHub updates.

Stripped the build commits as you requested and rebased.
